jina/types/document/mixins/attribute.py

```diff
@@ -139,13 +139,15 @@ def update(
         self: T,
         source: T,
         fields: Optional[List[str]] = None,
+        exclude_fields: Optional[List[str]] = None,
```

Comment:
Do you use this anywhere in this PR?

Reply:
ah yeah I added it but then I no more need it. Just removed it from the PR

Reply:
> PR looks good! :) Can you maybe describe the reduce logic somewhere? I mean the business logic that Docs different to each other will be concatenated and Docs with same id are reduced in a way that the first Document will keep its properties if they are set in both?
> 
> I think the description can go either in the Mixin class Pydoc or in the function documentations?

I added a description to the Mixin docstring.
Do you reduce should also be documented in https://docs.jina.ai/fundamentals/document/documentarray-api/ ?
I'm thinking we just need to document it [here](https://docs.jina.ai/fundamentals/flow/topology/) at the end of [this PR](https://github.com/jina-ai/jina/pull/4030) because we don't expect the user to use reduce by himself, rather, should know how it is applied in the Flow by default